### PR TITLE
[PROD-1189] - set correct_rationale to correct value in case of empty optional field.

### PR DESCRIPTION
**Pull request by saadyousafarbi**
### Description
UBCPI (UBC Peer Instruction Tool for edX) is third-party package used by EDX for Peer instruction block. A recent ticket https://openedx.atlassian.net/browse/PROD-1173 was reported in which the display name was not being updated while editing this block. 
Upon investigation, it was noted that this third-party package expected a value (i.e. corrent_rationale)  at all times when updating/editing this block. In the reported case, the field Explanation which is OPTIONAL (responsible for correct_rationale value) was empty. As a result, whenever any field was updated on the block with this empty value, the block was not updated. 
This scenario occurs when we have a peer instruction block with the optional field (because n/a correct answer selected) and the course is exported and then imported back in.

### Sandbox
https://studio-saadyousafarbi.sandbox.edx.org/course/course-v1:edX+DemoX+Demo_Course

### Reviewers
If you've been tagged for review, please check your corresponding box once you've given the :+1:.
- [ ] @awaisdar001 
- [x] @DawoudSheraz 

### Post-review
- [ ] Rebase and squash commits

### CODE REVIEW

**Review by DawoudSheraz — COMMENTED** [review]
1. Add a test
2. Run the tests locally and add the output screenshot as the tests aren't running on the PR.

**Comment by saadyousafarbi** [comment]
Screenshot for python and js tests that were run after adding the new test.
### JS tests
js
### Python tests
python_tests_ubcpi

File: ubcpi/static/js/spec/ubcpi_edit_spec.js [hunk 1]
```diff
@@ -381,7 +381,76 @@ describe('UBCPI_Edit module', function () {
                 expect(mockNotify.calls.count()).toBe(3);
             });
         });
-    })
+    });
+    describe('EditSettingsController', function () {
```
Comment (DawoudSheraz) [inline ubcpi/static/js/spec/ubcpi_edit_spec.js]: Is there not any existing test suite w.r.t editing? Writing an entire suite for an edge case is kind-of overkill.

Comment (saadyousafarbi) [reply ubcpi/static/js/spec/ubcpi_edit_spec.js]: Yes, there is. But for our case, we need to set up the mock data differently (without **correct_rationale** value) to cover our case. We are only checking the initial state of the controller in this test suite. The existing suite has other tests as well.

File: ubcpi/static/js/spec/ubcpi_edit_spec.js [hunk 2]
```diff
@@ -381,7 +381,76 @@ describe('UBCPI_Edit module', function () {
...
+            createController = function (params) {
+                return $controller(
+                    'EditSettingsController', {
+                        $scope: $rootScope,
+                        $stateParams: params || {}
+                    });
+            }
+        }));
+
+        beforeEach(function() {
```
Comment (DawoudSheraz) [inline ubcpi/static/js/spec/ubcpi_edit_spec.js]: Why are there two `beforeEach` definitions?

Comment (saadyousafarbi) [reply ubcpi/static/js/spec/ubcpi_edit_spec.js]: This can be done in a single `beforeEach` definition but for better readability and staying consistent with the existing similar test_suite, there are two definitions. The definitions are called from top to bottom.

**Comment by DawoudSheraz** [comment]
@saadyousafarbi Have you created a sandbox? If not, please do so to move this review further.

File: ubcpi/static/js/spec/ubcpi_edit_spec.js (hunk 2 above)
Comment (DawoudSheraz) [reply ubcpi/static/js/spec/ubcpi_edit_spec.js]: They should be moved to one. This doesn't affect readability but adds a sense of ambiguity as a function has been defined twice with different content.

**Review by DawoudSheraz — APPROVED** [review]
Verified on sandbox. Please squash your commits before merging

**Review by awaisdar001 — APPROVED** [review]
👍 Looks good to me Saad!
